src/main/kotlin/com/github/biomejs/intellijbiome/extensions/FileExt.kt

```diff
@@ -30,11 +32,13 @@ fun VirtualFile.isBiomeConfigFile(): Boolean =
     configValidExtensions.map { "${configName}.$it" }.contains(this.name)
 
 fun VirtualFile.findNearestBiomeConfig(root: VirtualFile? = null): VirtualFile? =
-    this.findNearestFile(root) {
-        if (it.isBiomeConfigFile()) {
-            // Skip biome.json(c) files that includes `root: false` or `extends: //`
-            BiomeConfig.loadFromFile(it)?.isRootConfig() ?: false
-        } else {
-            false
-        }
-    }
+  this.findNearestFile(root) { file ->
+    // 1. Ignore files that are not Biome config files
+    if (!file.isBiomeConfigFile()) return@findNearestFile false
+
+    // 2. If called with a 'root' (submodule), accept the first one found
+    if (root != null) return@findNearestFile true
```

Comment:
This isn't currect, `root` is always NOT null unless we can't find the project root. This changes breaks the new-style monorepo support and launches a Biome instance for each package. I think we should use the child config only when we have searched to the project root and no root config was found.

<img width="407" height="238" alt="image" src="https://github.com/user-attachments/assets/bbec8ad8-99cf-44f5-9e57-740826b62121" />